web/src/app/administration/components/workbasket-list-toolbar/workbasket-list-toolbar.component.ts

```diff
@@ -61,20 +46,16 @@ export class WorkbasketListToolbarComponent implements OnInit {
     });
   }
 
-  sorting(sort: Sorting) {
+  sorting(sort: Sorting<WorkbasketQuerySortParameter>) {
     this.performSorting.emit(sort);
   }
 
-  filtering(filterBy: Filter) {
-    if (this.component === 'workbasket-list') {
-      this.performFilter.emit(filterBy);
+  filtering({ left: component, right: filter }: Pair<string, WorkbasketQueryFilterParameter>) {
```

Comment:
instead of left, right maybe component, filterParams? Otherwise you have a reason why left, right for Pairs is better idea